### PR TITLE
fix: Update github actions versions

### PR DESCRIPTION
**Pull request by raizo07**
Closes [1320](https://github.com/software-mansion/starknet.py/issues/1320)

## Introduced changes
Update github actions versions

##

- [x] This PR contains breaking changes

- [Tests on networks (testnet)](https://github.com/software-mansion/starknet.py/actions/runs/11142932489/job/30966952153#logs)

### CODE REVIEW

**Comment by raizo07** [comment]
@franciszekjob Kindly review.

**Comment by raizo07** [comment]
@franciszekjob any update on this?

**Comment by raizo07** [comment]
@franciszekjob Updated, Kindly review.

**Comment by raizo07** [comment]
@franciszekjob  any update on this?

File: .github/workflows/checks.yml [hunk 1]
```diff
@@ -528,9 +535,9 @@ jobs:
...
-      - name: Generate coverage in XML
-        run: |
+          - name: Generate coverage in XML
+            run: |
           poetry run coverage xml
 
-      - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+          - name: Upload coverage to Codecov
+            uses: codecov/codecov-action@v4
```
Comment (franciszekjob) [inline .github/workflows/checks.yml:543]: Please remove formatting change and add newline 👍 .

**Comment by franciszekjob** [comment]
@raizo07 just reviewed 👍

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -241,7 +237,9 @@ jobs:
           poetry run coverage xml
 
       - name: Upload coverage to Codecov
-        uses: codecov/codecov-action@v3
+        uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}
```
[the remark this answers is not in the document]
Comment (raizo07) [reply .github/workflows/checks.yml:242]: Yes V4 does, previously it was not needed for public repos in V3, but for v4 it is needed for both public and private repos @franciszekjob

**Comment by raizo07** [comment]
@franciszekjob Kindly review

**Comment by franciszekjob** [comment]
@raizo07 you didn't apply changes I mentioned in [previous comment](https://github.com/software-mansion/starknet.py/pull/1504#pullrequestreview-2363106405)

**Pull request closed without merging**